### PR TITLE
Added version check for neoforge filter

### PR DESCRIPTION
**Pull request by Trial97**
Made the neoforge loaders filter dependent on the instance version.
Currently, 1.20.1 is compatible with the forge, but with 1.20.2 the compatibility is dropped.

### CODE REVIEW

**Review by getchoo — APPROVED** [review]

**Comment by TheKodeToad** [comment]
I don't think this version comparison works with snapshots (and I wouldn't actually be surprised if they are supported in future at this point lol)
Maybe just drop the `|= Forge` entirely?

**Comment by Trial97** [comment]
It looks like that would not be a problem the Version class can handle the snapshots stuff.
So to make sure I added a test for that.
Let's consider the drop of automatic Forge support when we implement loaders filter overload.

**Comment by TheKodeToad** [comment]
I'm talking about the [year]w[week][a-z] format

**Comment by Trial97** [comment]
Ok, I added a more strict condition: the forge is added just for 1.20.1.
Hope that we implement the filter thing before we need more changes here.

File: launcher/minecraft/PackProfile.cpp [hunk 1]
```diff
@@ -1018,8 +1018,7 @@ std::optional<ModPlatform::ModLoaderTypes> PackProfile::getSupportedModLoaders()
     // TODO: remove this or add version condition once Quilt drops official Fabric support
     if (loaders & ModPlatform::Quilt)
         loaders |= ModPlatform::Fabric;
-    // TODO: remove this or add version condition once NeoForge drops official Forge support
-    if (loaders & ModPlatform::NeoForge)
+    if (getComponentVersion("net.minecraft") == "1.20.1" && loaders & ModPlatform::NeoForge)
```
Comment (TheKodeToad) [inline launcher/minecraft/PackProfile.cpp]:
```suggestion
    if (getComponentVersion("net.minecraft") == "1.20.1" && (loaders & ModPlatform::NeoForge))
```

**Review by TheKodeToad — APPROVED** [review]

**Pull request merged**